.github/workflows/release-please.yml

```diff
@@ -0,0 +1,93 @@
+on:
+  push:
+    branches:
+      - main
+
+name: Run Release Please
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v3
+        id: release
+        with:
+          command: manifest
+          token: ${{ secrets.RELEASE_PLEASE_ACTION_TOKEN }}
+          default-branch: main
+          signoff: "OpenFeature Bot <109696520+openfeaturebot@users.noreply.github.com>"
+    outputs:
+      release_created: ${{ steps.release.outputs.releases_created }}
+      all: ${{ toJSON(steps.release.outputs) }}
+      paths_released: ${{ steps.release.outputs.paths_released }}
+
+  sbom:
+    needs: release-please
+    runs-on: ubuntu-latest
+    if: ${{ fromJSON(needs.release-please.outputs.paths_released)[0] != null }}
+    continue-on-error: true
+    strategy:
+      matrix:
+        release: ${{ fromJSON(needs.release-please.outputs.paths_released) }}
+    env:
+      TAG: ${{ fromJSON(needs.release-please.outputs.all)[format('{0}--tag_name', matrix.release)] }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+      - name: Generate SBOM
+        run: |
+          npm ci
+          if [ "${{ matrix.release }}" != "" ] && [ "${{ matrix.release }}" != "." ]; then
+            npm sbom --sbom-format=cyclonedx --omit=dev --omit=peer --workspace="${{ matrix.release }}" > bom.json
+          else
+            npm sbom --sbom-format=cyclonedx --omit=dev --omit=peer > bom.json
+          fi
+      - name: Attach SBOM to artifact
+        env:
+          GITHUB_TOKEN: ${{ secrets.RELEASE_PLEASE_ACTION_TOKEN }}
+        run: gh release upload $TAG bom.json
+
+  npm-release:
+    needs: release-please
+    runs-on: ubuntu-latest
+    if: ${{ needs.release-please.outputs.release_created }}
+    environment: publish
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          registry-url: "https://registry.npmjs.org"
+          cache: "npm"
+      - name: Update npm (for OIDC support)
```

Comment:
Updated this script a bit to rely on OIDC connection for NPM: https://docs.npmjs.com/trusted-publishers

Reply:
This is necessary since you're added `packageManager` in the package json? Won't the setup node action handle installing the correct version?

Reply:
yea AI was wrong, `corepack` doesn't update NPM, only Yarn / pnpm. reverted to: `npm install -g npm@^11.5.1`